cpp/dolfinx/mesh/graphbuild.cpp

```diff
@@ -364,42 +364,61 @@ mesh::build_local_dual_graph(CellType celltype,
             0, std::vector<std::int32_t>()};
   }
 
-  int tdim = mesh::cell_dim(celltype);
-  int num_cell_vertices = mesh::cell_num_entities(celltype, 0);
-  const std::int32_t num_cells = cells.size() / num_cell_vertices;
-
-  // Note: only meshes with a single cell type are supported
-  const graph::AdjacencyList<int> cell_facets
-      = mesh::get_entity_vertices(celltype, tdim - 1);
+  assert(cells.size() == celltypes.size());
 
-  // Determine maximum number of vertices for facet
+  // Create indexing offset for each cell type
+  // and determine max number of vertices per facet
+  std::vector<std::int32_t> cell_offsets = {0};
   int max_vertices_per_facet = 0;
-  for (int i = 0; i < cell_facets.num_nodes(); ++i)
+  int facet_count = 0;
+  int tdim = mesh::cell_dim(celltypes.front());
+  for (std::size_t j = 0; j < cells.size(); ++j)
   {
-    max_vertices_per_facet
-        = std::max(max_vertices_per_facet, cell_facets.num_links(i));
+    assert(tdim == mesh::cell_dim(celltypes[j]));
+    int num_cell_vertices = mesh::cell_num_entities(celltypes[j], 0);
+    std::int32_t num_cells = cells[j].size() / num_cell_vertices;
+    cell_offsets.push_back(cell_offsets.back() + num_cells);
+    facet_count += mesh::cell_num_entities(celltypes[j], tdim - 1) * num_cells;
+
+    graph::AdjacencyList<int> cell_facets
```

Comment:
Is `int` the appropriate type? Or is it `std::Int32_t`? (same in practice, but latter 'says' more).